gdk/wizard/Prompter.py

```diff
@@ -0,0 +1,259 @@
+from gdk.wizard.WizardData import WizardData
+from gdk.wizard.ConfigEnum import ConfigEnum
+from gdk.wizard.WizardChecker import WizardChecker
+from gdk.wizard.WizardConfigUtils import WizardConfigUtils
+from PyInquirer import prompt
+import argparse
+import sys
+
+
+class Wizard:
+    """
+    A class used to represent the GDK Startup Wizard
+    """
+
+    def __init__(self) -> None:
+        """
+        Initialize the Wizard object
+
+        Attributes
+        ----------
+            data: Wizard_data object
+            model: Wizard getter and setter object
+            checker: Wizard_checker object
+            parser: argparse object
+        """
+
+        self.utils = WizardConfigUtils()
+        self.project_config_file = self.utils.get_project_config_file()
+        self.field_dict = self.utils.read_from_config_file()
+
+        self.data = WizardData(self.field_dict)
+        self.checker = WizardChecker()
+        self.parser = argparse.ArgumentParser()
+
+    def prompter(self, field, required, max_attempts=3):
+        """
+        Prompts the user for a value of a given field key
+
+        Parameters
+        ----------
+            field (enum): a field key of the gdk_config file to be changed
+            value (string): the current value corresponding the field key to be changed
+            required (boolean): if the field is a required field of the gdk config file
+            max_attempts (int): the maximum number of attempts to get a valid response from the user
+
+        Returns
+        -------
+            string: the value for the field key "field"
+        """
+        current_field_value = self.data.get_field(field)
+        require = "REQUIRED " if required else "OPTIONAL "
+        link = "https://docs.aws.amazon.com/greengrass/v2/developerguide/gdk-cli-configuration-file.html#gdk-config-format"
+        for attempt in range(1, max_attempts + 1):
+            parser_argument = field.value.key
+            if field == ConfigEnum.BUILD_OPTIONS:
+                parser_argument = "build_options"
+            elif field == ConfigEnum.PUBLISH_OPTIONS:
+                parser_argument = "publish_options"
+
+            args = self.parser.parse_args(
+                [
+                    f"--{parser_argument}",
+                    self.interactive_prompt(
+                        parser_argument, current_field_value, require
+                    ),
+                ]
+            )
+            response = getattr(args, parser_argument).strip()
+
+            # only way customer is asked about custom_build_command if they have custom build system
+            # in which case they must specify a custom build command that is not None
+            if field == ConfigEnum.CUSTOM_BUILD_COMMAND:
+                if response == "None":
```

Comment:
What happens when the input is an empty string on an empty [] object?

Reply:
I see, I forgot to account for that. My thought is to also check len(response) < 1 what do you think of that

Reply:
Add this validation to the checker file is_valid_custom_build_command.

Reply:
You can then remove L72-L77.